core/src/db/seat_assignment/base_seat_assignment.entity.ts

```diff
@@ -0,0 +1,12 @@
+import { BaseEntity } from '../base.entity';
+import { SeatEntity } from '../seat/seat.entity';
+import { OneToOne } from 'typeorm';
+import { UserEntity } from '../user/user.entity';
+
+export abstract class BaseSeatAssignmentEntity extends BaseEntity {
+	@OneToOne(() => SeatEntity)
+	seat: Promise<SeatEntity>;
+
+	@OneToOne(() => UserEntity)
+	user: Promise<UserEntity>;
```

Comment:
I'm still not convinced that User is always correct here. We can revisit later, though.